frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -71,83 +73,114 @@ export type Hyperparameter = {
-    }
-
-    let context: any = z.preprocess(Number, number_context);
-    if (!required) {
-        context = context.optional().or(z.literal(''));
-    }
-    return context;
+const positiveIntValidator = (fieldName: string, required = true, minimum = 1, inputProps : rangeValidatorProps = {}) => {
+    return numberValidator(fieldName, 
+        {
+            min: minimum,
+            required: required,
+            ...inputProps
+        })
 };
 
-const positiveIntMessage = (field: string, required = true) =>
-    `${field} ${required ? 'is required and ' : ''}must be a positive integer`;
+type rangeValidatorProps = {
+    min?: number;
+    max?: number;
+    isFloat?: boolean;
+    required?: boolean;
+    allowAuto?: boolean;
+    includeMin?: boolean;
+    includeMax?: boolean;
+    alternateValues?: string[];
+}
 
-const positiveIntValidator = (fieldName: string, required = true, minimum = 1) => {
-    const number_context = z.number().min(minimum, {
-        message: `${positiveIntMessage(fieldName, required)} of at least ${minimum}`,
-    });
+/**
+ * Creates a validator for ensuring a number meets the required conditions
+ * 
+ * @param fieldName Name of the field being validated
+ * @param inputProps Properties that ensure the validity of the field
+ * @returns A validator for the field that ensures the desired requirements
+ */
+const numberValidator = (
+    fieldName: string,
+    inputProps?: rangeValidatorProps
+) => {
+    // Create properties by merging the defaults with the provided input properties
+    const props = {
+        min: Number.NEGATIVE_INFINITY, // Specifies the smallest number allowed (default is infinitely negative to allow for any number)
+        max: Number.POSITIVE_INFINITY, // Specifies the largest number allowed (default is ifinitely positive to allow for any number)
+        isFloat: false, // The number is assumed to be an Integer unless specified (default is that it is an Integer)
+        required: false, // Specify whether this field is required for submission and must not be empty (default is that the field is not required)
+        allowAuto: false, // Allows for the text option of "auto" instead of a number (default is that "auto" is not valid)
+        includeMin: true, // Whether the specified minimum is a valid number option (default is that it is valid)
+        includeMax: true, // Whether the specified maximum is a valid number option (default is that it is valid)
+        ...inputProps
+    }
 
-    let context: any = z.preprocess(Number, number_context);
-    if (!required) {
-        context = context.optional().or(z.literal(''));
+    // Create range message
+    let rangeMessage = '';
+    if (props.min !== Number.NEGATIVE_INFINITY && props.max !== Number.POSITIVE_INFINITY){
+        // Create range message
+        rangeMessage = ` value in the range ${props.includeMin?'[':'('}${props.min},${props.max}${props.includeMax?']':')'}`
```

Comment:
Similar comment to the above if the linter isn't complaining about the lack of spaces between in your ternary then we probably need to update the configuration. In the meantime just manually add the spaces.